backend/dataall/modules/dataset_sharing/services/data_sharing_service.py

```diff
@@ -218,3 +200,63 @@ def revoke_share(cls, engine: Engine, share_uri: str):
             share_sm.update_state(session, share, new_share_state)
 
             return revoked_tables_succeed and revoked_folders_succeed
+
+    @staticmethod
+    def create_lf_processor(session,
+                            dataset,
+                            share,
+                            shared_tables,
+                            revoked_tables,
+                            source_environment,
+                            target_environment,
+                            env_group):
+        try:
+            catalog_details = GlueClient(database=dataset.GlueDatabaseName,
+                                         account_id=source_environment.AwsAccountId,
+                                         region=source_environment.region).get_source_catalog()
+
+            source_account_id = catalog_details.account_id if catalog_details else source_environment.AwsAccountId
+
+            if source_account_id != target_environment.AwsAccountId:
+                processor = ProcessLFCrossAccountShare(
+                    session,
+                    dataset,
+                    share,
+                    shared_tables,
+                    revoked_tables,
+                    source_environment,
+                    target_environment,
+                    env_group,
+                    catalog_details
+                )
+            else:
+                processor = ProcessLFSameAccountShare(
+                    session,
+                    dataset,
+                    share,
+                    shared_tables,
+                    revoked_tables,
+                    source_environment,
+                    target_environment,
+                    env_group,
+                )
+            return processor
+        except Exception as e:
+            log.error(f"Error creating LF processor: {e}")
+            for table in shared_tables:
+                DataSharingService._handle_table_share_failure(session, share, table, ShareItemStatus.Share_Approved.value)
+            for table in revoked_tables:
+                DataSharingService._handle_table_share_failure(session, share, table, ShareItemStatus.Revoke_Approved.value)
+
+    @staticmethod
+    def _handle_table_share_failure(session, share, table, share_item_status):
```

Comment:
This seems to be new? It doesn't seem like it's modified code or related to central catalog support? What exact problem is this solving?

Reply:
As the `create_lf_processor` has many api calls with glue, there might be exceptions and this function handles and puts all the share_table_items to Failed state and doesn't leave them in a improper state ( Share_IN_Progress ) where they get stuck